Cargo.toml

```diff
@@ -69,5 +69,11 @@ rustc-workspace-hack = { path = 'src/tools/rustc-workspace-hack' }
 rustc-std-workspace-core = { path = 'src/tools/rustc-std-workspace-core' }
 rustc-std-workspace-alloc = { path = 'src/tools/rustc-std-workspace-alloc' }
 
+memoffset = { git = "https://github.com/ia0/memoffset", branch = "rust_issues_61053" }
```

Comment:
cc @rkruppe @RalfJung (the crate attempts `offsetof`)

Reply:
So we are now going to make rustc depend on some branch of some foreign git repo...?

Reply:
FWIW, this [uses `mem::uninitialized`](https://github.com/ia0/memoffset/blob/ee03d7a6d2504de16ad17daf744d8267de243736/src/offset_of.rs) so I don't even need to do a detailed review to know that it is UB.

Reply:
`offsetof` is unimplementable currently AFAIK. This is blocked on https://github.com/rust-lang/rfcs/pull/2582, which is blocked on the lang team.

Reply:
Those are existing dependencies. I just override them to be able to compile a stage2 compiler because those crates have errors (see the top commit of each of those branches). This will go away when I'll migrate from an error to an allow-by-default lint, because then we can use the released version without patching since there will be essentially none of the checks introduced by this PR. If I get time, I'll do this migration this week-end.

About why the compiler depends on memoffset, this is another story (unrelated to this PR), but I guess it's worth investigating if you believe it's problematic.

Reply:
I see, makes sense.

> About why the compiler depends on memoffset, this is another story (unrelated to this PR), but I guess it's worth investigating if you believe it's problematic.

Yeah, if we need `offsetof` in the compiler it should at least be implemented using `MaybeUninit`, not `mem::uninitialized`.  That's still wrong, but it's less wrong. ;)

 `MaybeUninit` will become stable on July 4th.